### PR TITLE
Minify HTML for the cached files

### LINKED ISSUE #33: Remove unnecessary whitespace & html comments
The saved files could be smaller without unnecessary whitespace & html comments.

### PR DESCRIPTION
**Pull request by benoitchantre**
Remove whitespaces and HTML comments (keep conditional comments).

Closes #33

### CODE REVIEW

File: inc/dropins/file-based-page-cache-functions.php [hunk 1]
```diff
@@ -68,8 +68,13 @@ function sc_cache( $buffer, $flags ) {
 
 	$modified_time = time(); // Make sure modified time is consistent
 
+	if ( ! WP_DEBUG ) {
```
Comment (tlovett1) [inline inc/dropins/file-based-page-cache-functions.php]: Need to check if `WP_DEBUG` is defined properly:

`if ( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG ) {`

**Comment by tlovett1** [comment]
Great idea.

**Comment by benoitchantre** [comment]
Thanks for the review. I have committed an improvement based on your comment.

**Comment by benoitchantre** [comment]
To be safer, it shouldn't remove whitespace inside `<pre>` and `<textarea>`, because browsers respect whitespace in these tags.

**Comment by tlovett1** [comment]
The more I think about this, the more I think it's a bad idea. What if there is an inline script that depends on more than one space? I just think there are too many edge cases that this would fully break.

**Comment by benoitchantre** [comment]
Maybe it could be an opt-in option you can activate when you have fully tested your website?
I have seen that the [PageSpeed filter](https://modpagespeed.com/doc/filter-whitespace-collapse) doesn't touch inline scripts, but edge cases can remain.
The savings are probably not worth the risks.

**Pull request closed without merging**